### PR TITLE
[dagster-powerbi] Update Power BI documentation to showcase new API patterns

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Updates our Power BI docs to incorporate the new specs-building pattern, norm of passing the resource to `Definitions`, and the new pattern for building executable asset definitions.

### CODE REVIEW

**Comment by benpankow** [comment]
* **#25468** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/integrations/powerbi.mdx [hunk 1]
```diff
@@ -30,19 +30,20 @@ This guide provides instructions for using Dagster with Power BI. Your Power BI
 
 To load Power BI assets into the Dagster asset graph, you must first construct a <PyObject module="dagster_powerbi" object="PowerBIWorkspace" /> resource, which allows Dagster to communicate with your Power BI workspace. You'll need to supply your workspace ID and credentials. You may configure a service principal or use an API access token, which can be passed directly or accessed from the environment using <PyObject object="EnvVar" />.
 
-Dagster can automatically load all semantic models, data sources, reports, and dashboards from your Power BI workspace. Call the <PyObject module="dagster_powerbi" object="PowerBIWorkspace" method="build_defs" /> function, which returns a <PyObject object="Definitions" /> object containing all the asset definitions for these Power BI assets.
+Dagster can automatically load all semantic models, data sources, reports, and dashboards from your Power BI workspace as asset specs. Call the <PyObject module="dagster_powerbi" method="load_powerbi_asset_specs" /> function, which returns a list of <PyObject object="AssetSpec" />s representing your Power BI assets. You can then include these asset specs in your <PyObject object="Definitions" /> object:
```
Comment (cmpadden) [inline docs/content/integrations/powerbi.mdx:33]: `load_powerbi_asset_specs` link does not work; if this is new make sure to run `make apidoc-build`.

image

File: docs/content/integrations/powerbi.mdx [hunk 2]
```diff
@@ -131,27 +145,34 @@ marketing_team_workspace = PowerBIWorkspace(
...
+        "marketing_power_bi": marketing_team_workspace,
+        "sales_power_bi": sales_team_workspace,
+    },
 )
 ```
 
 ## Materialize Power BI semantic models from Dagster
 
-Dagster's default behavior is to pull in representations of Power BI semantic models as external assets, which appear in the asset graph but can't be materialized. However, you can instruct Dagster to allow you to materialize these semantic models, refreshing them, by passing `enable_refresh_semantic_models=True` to the <PyObject module="dagster_powerbi" object="PowerBIWorkspace" method="build_defs" /> function:
+Dagster's default behavior is to pull in representations of Power BI semantic models as external assets, which appear in the asset graph but can't be materialized. However, you can build executable asset definitions that trigger the refresh of Power BI semantic models. The <PyObject module="dagster_powerbi" method="build_semantic_model_refresh_asset_definition" /> utility will construct an asset definition that triggers a refresh of a semantic model when materialized.
```
Comment (cmpadden) [inline docs/content/integrations/powerbi.mdx:163]: Broken link for this one too

**Review by cmpadden — CHANGES_REQUESTED** [review]
Overall looks great--just a few broken links possibly due to a needed `make apidoc-build`.

**Review by cmpadden — APPROVED** [review]
The passing of `specs` to the `assets` Definitions parameter still feels odd to me.

And the amount of boilerplate required to filter executable / non-executable assets is a bit much for end-users. But I'm liking the direction that these things are moving, and I suspect we'll be able to reduce the API surface over time.

Thanks for all the great work, and making sure docs are included @benpankow .

**Pull request merged**